modules/apps/transfer/keeper/grpc_query.go

```diff
@@ -81,3 +81,30 @@ func (q Keeper) Params(c context.Context, _ *types.QueryParamsRequest) (*types.Q
 		Params: &params,
 	}, nil
 }
+
+// DenomHash implements the Query/DenomHash gRPC method
+func (q Keeper) DenomHash(c context.Context, req *types.QueryDenomHashRequest) (*types.QueryDenomHashResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	// Convert given request trace path to DenomTrace struct to confirm the path in a valid denom trace format
+	denomTrace := types.ParseDenomTrace(req.Trace)
+	if err := denomTrace.Validate(); err != nil {
+		return nil, err
```

Comment:
should this be 
```suggestion
		return nil, status.Error(codes.InvalidArgument, err.Error())
```
?

Reply:
indeed, added suggestion